plugins/finder/jevents/jevents.php

```diff
@@ -289,28 +289,42 @@ protected function index(FinderIndexerResult $item, $format = 'html')
 		// get the data and query models
 		$dataModel = new JEventsDataModel();
 		$queryModel = new JEventsDBModel($dataModel);
+
 		// get the repeat (allowing for it to be unpublished)
-		$theevent = $queryModel->listEventsById($item->rp_id);
-		
-		if ($this->params->get("future", -1) != -1 && $theevent)
-		{
-			$past = $this->params->get("past", -1);
-			$date = new JDate($theevent->startDate() . " - $past days");
-			$item->publish_start_date = $date->toSql();
-		}
-		else
-		{
-			$item->publish_start_date	= isset($item->modified) ?$item->modified : "2010-01-01 00:00:00" ;			
-		}
-		if ($this->params->get("past", -1) != -1  && $theevent)
-		{
-			$future = $this->params->get("future", -1);
-			$date = new JDate($theevent->endDate() . " + $future days");
-			$item->publish_end_date = $date->toSql();			
+		$theevent = array($queryModel->listEventsById($item->rp_id));
+
+		JPluginHelper::importPlugin('jevents');
+		$dispatcher = JEventDispatcher::getInstance();
+		$dispatcher->trigger('onJevFinderIndexing', array(&$theevent));
+
+		$theevent = count($theevent) === 1 ? $theevent[0] : $theevent;
```

Comment:
does onJevFinderIndexing change $theevent from an array to a stdClass?  If not then the next couple of lines will throw an error

Reply:
Yep as per the JEvents  Bit Timelimit plugin PR, we implement a new function and run them through already available code in the timelimit plugin.